common/vm_cleanup_snapshot.yml

```diff
@@ -51,12 +51,13 @@
         - include_tasks: vm_revert_snapshot.yml
           vars:
             snapshot_name: "{{ revert_to_snapshot_name }}"
-      when: revert_to_snapshot_name
+      when:
+        - revert_to_snapshot_name
+        - current_snapshot_name != revert_to_snapshot_name
```

Comment:
I suggest removing `current_snapshot_name != revert_to_snapshot_name` as even though the VM current snapshot name is same as the snapshot to revert, the VM could have been changed, and its state is not exactly same as state of current snapshot.

Reply:
Make sense. Fixed.